pkg/proxy/iptables/proxier.go

```diff
@@ -1510,19 +1506,28 @@ func (proxier *Proxier) syncProxyRules() {
 	metrics.IptablesRulesTotal.WithLabelValues(string(utiliptables.TableFilter)).Set(float64(proxier.filterRules.Lines()))
 	metrics.IptablesRulesTotal.WithLabelValues(string(utiliptables.TableNAT)).Set(float64(proxier.natRules.Lines()))
 
-	// Write the end-of-table markers.
-	proxier.filterRules.Write("COMMIT")
-	proxier.natRules.Write("COMMIT")
-
 	// Sync rules.
-	// NOTE: NoFlushTables is used so we don't flush non-kubernetes chains in the table
 	proxier.iptablesData.Reset()
+	proxier.iptablesData.WriteString("*filter\n")
 	proxier.iptablesData.Write(proxier.filterChains.Bytes())
 	proxier.iptablesData.Write(proxier.filterRules.Bytes())
+	proxier.iptablesData.WriteString("COMMIT\n")
+	proxier.iptablesData.WriteString("*nat\n")
 	proxier.iptablesData.Write(proxier.natChains.Bytes())
 	proxier.iptablesData.Write(proxier.natRules.Bytes())
-
+	proxier.iptablesData.WriteString("COMMIT\n")
+
+	klog.V(2).InfoS("Reloading service iptables data",
+		"numServices", len(proxier.serviceMap),
+		"numEndpoints", proxier.endpointChainsNumber,
```

Comment:
this is capped to `endpointChainsNumberThreshold` , should we mention to avoid confusion of people reading the logs and seeing this value doesn't increase?

Reply:
endpointChainsNumberThreshold just governs whether iptables `--comment` is used or not?

Reply:
yeah, but if I'm debugging something with lot of services and I see that this number keeps being always 1k despite I know it should be 4k, I will be exploring this path.
I'm afraid this can be a red herring for a lot of admins

Reply:
I feel like I am missing something - `endpointChainsNumberThreshold` is used for exactly one thing, deciding whether to add comments or not.  `endpointChainsNumber` is accumulated with no cap.  What am I missing?

Reply:
lol, I thought the metrics was capped, my bad